loginCore/src/main/java/com/hamdan/forzenbook/login/core/viewmodel/BaseLoginViewModel.kt

```diff
@@ -131,14 +145,21 @@ abstract class BaseLoginViewModel(
         }
     }
 
-    fun loginClicked() {
+    fun loginClicked(context: Context) {
         if (loginState.getContent() is LoginContent.Code) {
-            submitLogin()
+            submitLogin(context)
         } else {
             requestLoginValidationCode()
         }
     }
 
+    private fun saveTokenToPreferences(context: Context, token: String) {
+        context.getSharedPreferences(TOKEN_PREFERENCE_LOCATION, Context.MODE_PRIVATE).edit().apply {
```

Comment:
Why is this here and also in Room? It should just be in preferences

Reply:
That is my mistake, I was thinking shared preferences was not persistent, I went and looked it up and see my mistake, if I'm holding it in shared preferences theres no reason to hold it in room as well. Will remove.